gobblin-utility/src/main/java/org/apache/gobblin/util/ClustersNames.java

```diff
@@ -97,21 +110,34 @@ public void addClusterMapping(URL clusterUrl, String clusterName) {
     this.urlToNameMap.put(clusterUrl.toString(), clusterName);
   }
 
-  // Strip out the port number if it is a valid URI
-  private static String normalizeClusterUrl(String clusterIdentifier) {
+  private static List<String> generateUrlMatchCandidates(String clusterIdentifier) {
+    ArrayList<String> candidates = new ArrayList<>();
+    candidates.add(clusterIdentifier);
+
     try {
       URI uri = new URI(clusterIdentifier.trim());
-      // URIs without protocol prefix
-      if (!uri.isOpaque() && null != uri.getHost()) {
-        clusterIdentifier = uri.getHost();
+      if (uri.getHost() != null) {
+        if (uri.getPort() != -1) {
+          candidates.add(uri.getHost() + ":" + uri.getPort());
+        }
+
+        candidates.add(uri.getHost());
```

Comment:
I don't see any chance for host:port to be picked here since getHost will always be placed after that. Is this intentional ? Or can you give an unit test case to return host:port

Reply:
Added a comment here. Also check out "testPortSpecificOverrides" test